components/nexus-bootstrap/src/main/java/org/sonatype/nexus/bootstrap/jetty/ConnectorManager.java

```diff
@@ -0,0 +1,262 @@
+  {
+    final HttpScheme httpScheme = connectorConfiguration.getScheme();
+    verifyConfiguration(httpScheme);
+
+    final HttpConfiguration httpConfiguration =
+        connectorConfiguration.customize(defaultHttpConfiguration(httpScheme));
+    final ServerConnector connectorPrototype =
+        defaultConnector(httpScheme);
+    final ServerConnector serverConnector;
+    if (HttpScheme.HTTP == httpScheme) {
+      serverConnector = new ServerConnector(
+          server,
+          connectorPrototype.getAcceptors(),
+          connectorPrototype.getSelectorManager().getSelectorCount(),
+          new InstrumentedConnectionFactory(
+              new HttpConnectionFactory(httpConfiguration)
+          )
+      );
+    }
+    else if (HttpScheme.HTTPS == httpScheme) {
+      final SslContextFactory sslContextFactory = bean(SSL_CONTEXT_FACTORY_ID, SslContextFactory.class);
+      serverConnector = new ServerConnector(
+          server,
+          connectorPrototype.getAcceptors(),
+          connectorPrototype.getSelectorManager().getSelectorCount(),
+          new InstrumentedConnectionFactory(
+              new SslConnectionFactory(sslContextFactory, "http/1.1")
+          ),
+          new HttpConnectionFactory(httpConfiguration)
+      );
+    }
+    else {
+      throw new UnsupportedHttpSchemeException(httpScheme);
+    }
+    serverConnector.setHost(connectorPrototype.getHost());
+    serverConnector.setPort(connectorConfiguration.getPort());
+    serverConnector.setIdleTimeout(connectorPrototype.getIdleTimeout());
+    serverConnector.setSoLingerTime(connectorPrototype.getSoLingerTime());
+    serverConnector.setAcceptorPriorityDelta(connectorPrototype.getAcceptorPriorityDelta());
+    serverConnector.setSelectorPriorityDelta(connectorPrototype.getSelectorPriorityDelta());
+    serverConnector.setAcceptQueueSize(connectorPrototype.getAcceptQueueSize());
+
+    managedConnectors.put(connectorConfiguration, serverConnector);
+    server.addConnector(serverConnector);
+    try {
+      serverConnector.start();
+    }
+    catch (Exception e) {
+      throw new RuntimeException(e); // NOSONAR: no guava in scope
+    }
+
+    return serverConnector;
+  }
+
+  /**
+   * Stops and removes the connector configuration from Jetty.
+   */
+  public void removeConnector(final ConnectorConfiguration connectorConfiguration) {
+    final ServerConnector serverConnector = managedConnectors.remove(connectorConfiguration);
+    if (serverConnector != null) {
+      try {
+        serverConnector.stop();
+      }
+      catch (Exception e) {
+        throw new RuntimeException(e); // NOSONAR: no guava in scope
+      }
+      server.removeConnector(serverConnector);
+    }
+  }
+
+  // ==
+
+  /**
+   * Verifies all the needed bits are present in Jetty XML configuration (as HTTPS must be enabled by users).
+   */
+  private void verifyConfiguration(final HttpScheme httpScheme) {
+    try {
```

Comment:
As with above comment, if there were classes representing supported schemes, this validation logic would be in that class, instead of having parallel if/else constructions.  (Then again, maybe we'll never, ever add another scheme.)